ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using ServerCore.DataModel;
+using ServerCore.Models;
+
+namespace ServerCore.ModelBases
+{
+    public abstract class EventSpecificPageModel : PageModel
+    {
+        [FromRoute]
+        [ModelBinder(typeof(EventBinder))]
+        public Event Event { get; set; }
+
+        public class EventBinder : IModelBinder
+        {
+            public async Task BindModelAsync(ModelBindingContext bindingContext)
+            {
+                string eventIdAsString = bindingContext.ActionContext.RouteData.Values["eventId"] as string;
+
+                if (int.TryParse(eventIdAsString, out int eventId))
+                {
+                    var puzzleServerContext = bindingContext.HttpContext.RequestServices.GetService<PuzzleServerContext>();
+                    Event eventObj = await puzzleServerContext.Events.Where(e => e.ID == eventId).FirstOrDefaultAsync();
+
+                    if (eventObj != null)
```

Comment:
If the lookup failed, what happens? I guess I'd hope for a 404

Reply:
So far no such luck. I can't figure out how to have the property be required.

Reply:
By which I mean that if I set the event ID to be out of scope, it just merrily continues along with Event == null. The binding result is {Failed}, but the model doesn't care.